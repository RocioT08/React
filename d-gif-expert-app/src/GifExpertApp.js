import React, { useState } from "react";
import AddCategory from "./components/AddCategory";
import GiftGrid from "./components/GiftGrid";

const categoriesTemplate = ["action"];

const GifExpertApp = ({ defaultCategories = [] }) => {
  const [categories, setCategories] = useState(defaultCategories);

  const addCategoryHandler = () => {
    setCategories([...categories, `category_${Math.random()}`]);
  };

  return (
    <>
      <h2>GifExpertApp</h2>
      <AddCategory setCategories={setCategories} />
      <hr />
      <button onClick={addCategoryHandler}>Add New</button>

      <ul>
        {categories.map((category) => (
          <GiftGrid key={category} category={category} />
        ))}
      </ul>
    </>
  );
};

export default GifExpertApp;
