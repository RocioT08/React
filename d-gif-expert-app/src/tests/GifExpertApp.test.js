import React from "react";
import { shallow } from "enzyme";
import "@testing-library/jest-dom";
import GifExpertApp from "../GifExpertApp";

describe("Test GifExpertApp Components", () => {
  test("should render the component correctly", () => {
    const wrapper = shallow(<GifExpertApp />);
    expect(wrapper).toMatchSnapshot();
  });

  test("should show a list of categories", () => {
    const categories = ["action", "drama"];
    const wrapper = shallow(<GifExpertApp defaultCategories={categories} />);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("GiftGrid").length).toBe(categories.length);
  });
});
