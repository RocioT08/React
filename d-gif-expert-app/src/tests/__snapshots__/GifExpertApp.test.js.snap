// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test GifExpertApp Components should render the component correctly 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <button
    onClick={[Function]}
  >
    Add New
  </button>
  <ul />
</Fragment>
`;

exports[`Test GifExpertApp Components should show a list of categories 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <button
    onClick={[Function]}
  >
    Add New
  </button>
  <ul>
    <GiftGrid
      category="action"
      key="action"
    />
    <GiftGrid
      category="drama"
      key="drama"
    />
  </ul>
</Fragment>
`;
